external/storm-kafka/src/jvm/org/apache/storm/kafka/PartitionManager.java

```diff
@@ -199,7 +199,12 @@ private void fill() {
         try {
             msgs = KafkaUtils.fetchMessages(_spoutConfig, _consumer, _partition, offset);
         } catch (TopicOffsetOutOfRangeException e) {
-            offset = KafkaUtils.getOffset(_consumer, _partition.topic, _partition.partition, kafka.api.OffsetRequest.EarliestTime());
+            long partitionLatestOffset = KafkaUtils.getOffset(_consumer, _partition.topic, _partition.partition, kafka.api.OffsetRequest.LatestTime());
+            if (partitionLatestOffset < offset) {
+                offset = partitionLatestOffset;
+            } else {
+                offset = KafkaUtils.getOffset(_consumer, _partition.topic, _partition.partition, kafka.api.OffsetRequest.EarliestTime());
+            }
             // fetch failed, so don't update the fetch metrics
             
             //fix bug [STORM-643] : remove outdated failed offsets
```

Comment:
@HeartSaVioR  Given that this code is inside TopicOffsetOutOfRangeException, which can only happen if the spout requested for an offset that's not in Kafka Topic. So in which case do you see that we might disregard the valid offsets.